services/web/server/src/simcore_service_webserver/socketio/messages.py

```diff
@@ -32,22 +32,37 @@
 SOCKET_IO_WALLET_OSPARC_CREDITS_UPDATED_EVENT: Final[str] = "walletOsparcCreditsUpdated"
 
 
-async def send_messages_to_user(
-    app: Application, user_id: UserID, messages: Sequence[SocketMessageDict]
-) -> None:
-    sio: AsyncServer = get_socket_server(app)
-
+async def _logged_gather_emit(
+    sio: AsyncServer,
+    *,
+    room: SocketIORoomStr,
+    messages: Sequence[SocketMessageDict],
+    max_concurrency: int = 100,
+):
     await logged_gather(
         *(
             sio.emit(
-                message["event_type"],
-                json_dumps(message["data"]),
-                room=SocketIORoomStr.from_user_id(user_id=user_id),
+                event=message["event_type"],
+                data=jsonable_encoder(message["data"]),
+                room=room,
             )
             for message in messages
         ),
         reraise=False,
         log=_logger,
+        max_concurrency=max_concurrency,
+    )
+
+
+async def send_messages_to_user(
+    app: Application, user_id: UserID, messages: Sequence[SocketMessageDict]
```

Comment:
just wondering now that I see this stuff with gather and not for now but for later:
- using gather here might change the order of the messages right? there is absolutely no control on what goes first, second, etc...

Reply:
I also share the same opinion as Sylvain here. The messages will arrive out of order and might not make sense for people any longer. I would still want to have this property of messages being sorted.